### DISCUSSION TITLE
关于游戏存档路径的提问

### DISCUSSION THREAD

**Discussion by mcthesw** [BeyondDimension/SteamTools#1400] [Q&A]
项目开发者们，你们好
我近期在开发一个游戏存档管理软件，但是未能找到获取Steam游戏存档路径、游戏信息的资料，所以没有办法做出自动识别本地Steam游戏的功能，在看到Steam++的新功能后我就来此求助了。
本来我不想打扰，但是Steam++的源代码对我来说有些庞大，无从下手阅读，所以我想询问一下能否简单和我描述一下获取游戏信息和存档位置的思路，以及是否有能参考的文档

**Comment by rmbadmin** [comment]
没怎么注意discussion才发现还有这么个提问，

识别游戏存档信息来源于steam的云存档功能，
从steam缓存信息和api获取其对应的游戏是否支持云存档以及云存档的信息来定位的游戏存档路径

**Reply by mcthesw** [reply]
感谢你的回复。
我刚才翻了翻steam云存档的路径，发现 userdata/SteamID/AppID/remotecache.vdf 这样的一个路径，该 vdf 文件可以使用纯文本编辑器打开，根据观察，内部应该存有游戏存档的路径。请问我这个思路是正确的吗？
还有一点，就是有关“steam缓存信息和api”，这点我没找到相关的信息，请问能给我提供更详细一些的信息吗，非常感谢！

**Reply by rmbadmin** [reply]
> 感谢你的回复。 我刚才翻了翻steam云存档的路径，发现 userdata/SteamID/AppID/remotecache.vdf 这样的一个路径，该 vdf 文件可以使用纯文本编辑器打开，根据观察，内部应该存有游戏存档的路径。请问我这个思路是正确的吗？ 还有一点，就是有关“steam缓存信息和api”，这点我没找到相关的信息，请问能给我提供更详细一些的信息吗，非常感谢！

Steam所有缓存游戏信息存在 /Steam/appcache/appinfo.vdf 
云存档相关api 则是 https://partner.steamgames.com/doc/api/ISteamRemoteStorage
你发的 userdata 的路径也确实与云存档相关，但应该只有相对路径，无法定位存档绝对路径
